compiler/optimizer/CompactLocals.cpp

```diff
@@ -90,6 +90,9 @@ int32_t TR_CompactLocals::perform()
       return 0;
       }
 
+   if (comp()->getProfilingMode() == JitProfiling)
```

Comment:
Unlike the other changes in this commit this seems to be a new optimization that has been disabled under JitProfiling (?)

Reply:
This optimization was disabled using strategy flags in the past. The initial version of this commit shifted these checks to the optimizations themselves, as they weren't part of the strategy, rather they were a restriction due to the profiling implementation.

The change is being reverted to use strategy flags, as this makes the disable clearer.